cluster/src/test/scala/org/apache/spark/sql/store/SnappyUDFTest.scala

```diff
@@ -472,9 +473,48 @@ class SnappyUDFTest extends SnappyFunSuite with BeforeAndAfterAll {
     snc.sql("insert into test123 values(87,76,63)")
     snc.sql("insert into test123 values(12,24,53)")
 
-    snc.sql("select DSID() from test123").collect().foreach(row=>{
+    snc.sql("select DSID() from test123").collect().foreach(row => {
       assert(row.getString(0).equals(Misc.getMyId().getId()))
     });
     snc.sql("drop table test123")
   }
+
+  test("Test UDF other schema") {
+    val conf = new ConnectionConfBuilder(snc.snappySession).build()
+    val conn = ConnectionUtil.getPooledConnection("test default conf", conf)
+    try{
+      val st = conn.createStatement
+      st.execute("create schema trade")
+      val udfText: String = "public class StringUDF implements" +
+          " org.apache.spark.sql.api.java.UDF1<String,String> {" +
+          " @Override public String call(String s){ " +
+          "               return s + s; " +
+          "}" +
+          "}"
+      val file = createUDFClass("StringUDF", udfText)
+      val jar = createJarFile(Seq(file))
+
+      snc.sql(s"CREATE FUNCTION TRADE.STRUDF AS StringUDF " +
+          s"RETURNS STRING USING JAR " +
+          s"'$jar'")
+
+      snc.sql("CREATE TABLE trade.rr_test_table(OrderRef INT NOT NULL, description String, " +
+          "price BIGINT, serviceTax DECIMAL, surcharge Float, purchase_date DATE, time Timestamp)")
+
+      snc.sql("CREATE TABLE trade.col_test_table(OrderRef INT NOT NULL," +
+          " description String, price  " +
+          "LONG, serviceTax DECIMAL, surcharge Float, purchase_date DATE, time Timestamp) " +
+          "using column options(PARTITION_BY 'OrderRef')")
+
+
+      snc.sql("select TRADE.strudf(description) from trade.col_test_table").collect()
```

Comment:
If I set the current schema as "TRADE" on the jdbcconn and then use the function without specifying the "schemaname" in the function, then will it work?